scripts/benchmark/checks/erc20_interact.py

```diff
@@ -0,0 +1,354 @@
+
+        if self.persist.deployed_contract_address is None:
+            self.persist.deployed_contract_address = self.deployed_contract_address
+
+        assert computation.is_success
+
+        self.simple_token = self.w3.eth.contract(
+            address=Web3.toChecksumAddress(encode_hex(self.deployed_contract_address)),
+            abi=self.contract_interface['abi'],)
+
+        if self.persist.simple_token is None:
+            self.persist.simple_token = self.simple_token
+
+    def erc_transfer(self, addr: str, chain: MiningChain) -> None:
+        ammout = self.num_blocks * self.num_tx
+
+        w3_tx1 = self.persist.simple_token.functions.transfer(
+            addr, ammout).buildTransaction(W3_TX_DEFAULTS)
+
+        tx1 = new_transaction(
+            vm=chain.get_vm(),
+            private_key=FUNDED_ADDRESS_PRIVATE_KEY,
+            from_=FUNDED_ADDRESS,
+            to=self.persist.deployed_contract_address,
+            amount=0,
+            gas=SECOND_TX_GAS_LIMIT,
+            data=decode_hex(w3_tx1['data']),
+        )
+
+        block, receipt, computation = chain.apply_transaction(tx1)
+
+        assert computation.is_success
+        assert to_int(computation.output) == 1
+
+    def erc_approve(self, addr2: str, chain: MiningChain) -> None:
+        ammout = self.num_blocks * self.num_tx
+
+        w3_tx2 = self.persist.simple_token.functions.approve(
+            addr2, ammout).buildTransaction(W3_TX_DEFAULTS)
+
+        tx2 = new_transaction(
+            vm=chain.get_vm(),
+            private_key=FUNDED_ADDRESS_PRIVATE_KEY,
+            from_=FUNDED_ADDRESS,
+            to=self.persist.deployed_contract_address,
+            amount=0,
+            gas=SECOND_TX_GAS_LIMIT,
+            data=decode_hex(w3_tx2['data']),
+        )
+
+        block, receipt, computation = chain.apply_transaction(tx2)
+
+        assert computation.is_success
+        assert to_int(computation.output) == 1
+
+    def erc_transfer_from(self, addr1: str, addr2: str, chain: MiningChain) -> None:
+
+        w3_tx3 = self.persist.simple_token.functions.transferFrom(
+            addr1, addr2, 1).buildTransaction(W3_TX_DEFAULTS)
+
+        tx3 = new_transaction(
+            vm=chain.get_vm(),
+            private_key=SECOND_ADDRESS_PRIVATE_KEY,
+            from_=SECOND_ADDRESS,
+            to=self.persist.deployed_contract_address,
+            amount=0,
+            gas=SECOND_TX_GAS_LIMIT,
+            data=decode_hex(w3_tx3['data']),
+        )
+
+        block, receipt, computation = chain.apply_transaction(tx3)
+
+        assert computation.is_success
+        assert to_int(computation.output) == 1
+
+
+class ERC20DeployBenchmark(BaseERC20Benchmark):
+    def __init__(self, persist: Persistant) -> None:
+        super().__init__(persist)
+
+    @property
+    def name(self) -> str:
+        return 'ERC20 deployment'
+
+    def _setup_benchmark(self, chain: MiningChain) -> None:
+        return
```

Comment:
Not looking into the actual code yet (because I guess it will change again to not reset the header). Just commenting on the code style: For an empty implementation you write `pass` instead of `return`